test_utils/src/sequencer.rs

```diff
@@ -204,23 +248,48 @@ where
 {
     async fn handle_messages(&self, mut stream: Stream) {
         let (tx_output, mut rx_output) = channel(100);
+        let subscriber_id = self.counter.fetch_add(1, Ordering::SeqCst);
 
         // Notify the core of a new subscriber.
         self.tx_subscriber
-            .send(tx_output)
+            .send(SubscriberMessage::Init(tx_output, subscriber_id))
             .await
             .expect("Failed to send new subscriber to core");
 
-        // Update the subscriber every time a certificate is sequenced.
-        while let Some((bytes, consensus_index)) = rx_output.recv().await {
-            let message = ConsensusOutput {
-                message: bytes.to_vec(),
-                sequencer_number: consensus_index,
-            };
-            let serialized = serialize_consensus_output(&message);
-            if stream.sink().send(Bytes::from(serialized)).await.is_err() {
-                log::debug!("Connection dropped by subscriber");
-                break;
+        // Interact with the subscriber.
+        loop {
+            tokio::select! {
+                // Update the subscriber every time a certificate is sequenced.
+                Some(message) = rx_output.recv() => {
+                    let serialized = serialize_consensus_output(&message);
```

Comment:
Advice: serialize these before you send them down the channel. And keep them as `Bytes` which has an efficient, no-copy, clone.